_install-and-configure/upgrade-opensearch/appendix/rolling-upgrade-lab.md

```diff
@@ -0,0 +1,887 @@
+   ```bash
+   curl -H "Content-Type: application/x-ndjson" \
+      -X PUT "https://localhost:9201/ecommerce?pretty" \
+      --data-binary "@ecommerce-field_mappings.json" \
+      -ku admin:admin
+   ```
+   {% include copy.html %}
+   <p class="codeblock-label">Example response</p>
+   ```json
+   {
+      "acknowledged" : true,
+      "shards_acknowledged" : true,
+      "index" : "ecommerce"
+   }
+   ```
+1. Use the [Bulk]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) API to add data to the new ecommerce index from `ecommerce.json`:
+   ```bash
+   curl -H "Content-Type: application/x-ndjson" \
+      -X PUT "https://localhost:9201/ecommerce/_bulk?pretty" \
+      --data-binary "@ecommerce.json" \
+      -ku admin:admin
+   ```
+   {% include copy.html %}
+   <p class="codeblock-label">Example response (truncated)</p>
+   ```json
+   {
+      "took" : 3323,
+      "errors" : false,
+      "items" : [
+   ...
+         "index" : {
+            "_index" : "ecommerce",
+            "_type" : "_doc",
+            "_id" : "4674",
+            "_version" : 1,
+            "result" : "created",
+            "_shards" : {
+               "total" : 2,
+               "successful" : 2,
+               "failed" : 0
+            },
+            "_seq_no" : 4674,
+            "_primary_term" : 1,
+            "status" : 201
+         }
+      ]
+   }
+   ```
+1. <p id="validation">A search query can also serve to confirm that the data was indexed successfully. The following query returns a count of documents where keyword `customer_first_name` equals `Sonya`:</p>
+   ```bash
+   curl -H 'Content-Type: application/json' \
+      -X GET "https://localhost:9201/ecommerce/_search?pretty=true&filter_path=hits.total" \
+      -d'{"query":{"match":{"customer_first_name":"Sonya"}}}' \
+      -ku admin:admin
+   ```
+   {% include copy.html %}
+   <p class="codeblock-label" id="query-validation">Example response</p>
+   ```json
+   {
+      "hits" : {
+         "total" : {
+            "value" : 106,
+            "relation" : "eq"
+         }
+      }
+   }
+   ```
+
+### Adding data using OpenSearch Dashboards
+
+1. Open a web browser and navigate to port `5601` on your Docker host (for example, <code>https://<var>HOST_ADDRESS</var>:5601</code>). If OpenSearch Dashboards is running, and you have network access to the host from your browser client, then you will be redirected to a login page.
+    1. If the web browser throws an error because the TLS certificates are self-signed, then you might need to bypass certificate checks in your browser. Refer to the browser's documentationn for information about bypassing certificate checks. The common name (CN) for each certficate is generated with respect to the container and node name for intra-cluster communication, so connecting to the host from a browser will still result in an "invalid CN" warning.
```

Comment:
I would format this as a line break versus a sub-ordered-list, since this sub-ordered-list only has one item. Alternatively, you find a way to break the item up into two items:

```suggestion
    If the web browser throws an error because the TLS certificates are self-signed, then you might need to bypass certificate checks in your browser. Refer to the browser's documentationn for information about bypassing certificate checks. The common name (CN) for each certficate is generated with respect to the container and node name for intra-cluster communication, so connecting to the host from a browser will still result in an "invalid CN" warning.
```